data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/S3SinkTest.java

```diff
@@ -0,0 +1,146 @@
+import org.opensearch.dataprepper.plugins.sink.codec.JsonCodec;
+import org.opensearch.dataprepper.plugins.sink.configuration.AwsAuthenticationOptions;
+import org.opensearch.dataprepper.plugins.sink.configuration.ObjectOptions;
+import org.opensearch.dataprepper.plugins.sink.configuration.ThresholdOptions;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+
+class S3SinkTest {
+
+    private S3SinkConfig s3SinkConfig;
+    private ThresholdOptions thresholdOptions;
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+    private AwsCredentialsProvider awsCredentialsProvider;
+    private ObjectOptions objectOptions;
+    private JsonCodec codec;
+    private S3Sink s3Sink;
+    private PluginSetting pluginSetting;
+    private PluginFactory pluginFactory;
+    private PluginModel pluginModel;
+
+    @BeforeEach
+    void setUp() {
+
+        s3SinkConfig = mock(S3SinkConfig.class);
+        thresholdOptions = mock(ThresholdOptions.class);
+        objectOptions = mock(ObjectOptions.class);
+        awsAuthenticationOptions = mock(AwsAuthenticationOptions.class);
+        codec = mock(JsonCodec.class);
+
+        pluginSetting = mock(PluginSetting.class);
+        pluginModel = mock(PluginModel.class);
+        pluginFactory = mock(PluginFactory.class);
+
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(thresholdOptions);
+        when(s3SinkConfig.getThresholdOptions().getEventCount()).thenReturn(100);
+        when(s3SinkConfig.getThresholdOptions().getMaximumSize()).thenReturn(ByteCount.parse("1kb"));
+        when(s3SinkConfig.getThresholdOptions().getEventCollectionDuration()).thenReturn(Duration.ofSeconds(5));
+
+        when(s3SinkConfig.getObjectOptions()).thenReturn(objectOptions);
+        when(s3SinkConfig.getObjectOptions().getFilePattern()).thenReturn("logs-${yyyy-MM-dd}");
+
+        when(s3SinkConfig.getTemporaryStorage()).thenReturn("local_file");
+        when(s3SinkConfig.getCodec()).thenReturn(pluginModel);
+
+        when(s3SinkConfig.getAwsAuthenticationOptions()).thenReturn(awsAuthenticationOptions);
+        when(awsAuthenticationOptions.getAwsRegion()).thenReturn(Region.of("us-east-1"));
+        when(awsAuthenticationOptions.authenticateAwsConfiguration()).thenReturn(awsCredentialsProvider);
+
+        when(pluginModel.getPluginName()).thenReturn("json");
+        when(pluginFactory.loadPlugin(Codec.class, pluginSetting)).thenReturn(codec);
+
+        when(pluginSetting.getName()).thenReturn("s3");
+        when(pluginSetting.getPipelineName()).thenReturn("S3-sink-pipeline");
+    }
+
+    @Test
+    void test_s3_Sink_plugin_isReady_ture() {
+        s3Sink = new S3Sink(pluginSetting, s3SinkConfig, pluginFactory);
+        s3Sink.doInitialize();
+        assertTrue(s3Sink.isReady(), "s3 sink is not ready to work");
+    }
+
+    @Test
+    void test_s3_Sink_plugin_isReady_false() {

```

Comment:
Nit. Names of the test functions should be camel case only. Do not use camel case and snake case both.